src/uu/ptx/src/ptx.rs

```diff
@@ -271,17 +281,40 @@ struct FileContent {
 
 type FileMap = HashMap<OsString, FileContent>;
 
-fn read_input(input_files: &[OsString]) -> std::io::Result<FileMap> {
+fn read_input(input_files: &[OsString], config: &Config) -> std::io::Result<FileMap> {
     let mut file_map: FileMap = HashMap::new();
     let mut offset: usize = 0;
+
+    let sentence_splitter = if let Some(re_str) = &config.sentence_regex {
+        Some(Regex::new(re_str).map_err(|e| {
+            std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                translate!("ptx-error-invalid-regexp", "error" => e),
+            )
+        })?)
+    } else {
+        None
+    };
+
     for filename in input_files {
-        let reader: BufReader<Box<dyn Read>> = BufReader::new(if filename == "-" {
+        let mut reader: BufReader<Box<dyn Read>> = BufReader::new(if filename == "-" {
             Box::new(stdin())
         } else {
             let file = File::open(Path::new(filename))?;
             Box::new(file)
         });
-        let lines: Vec<String> = reader.lines().collect::<std::io::Result<Vec<String>>>()?;
+
+        let lines = if let Some(re) = &sentence_splitter {
+            let mut buffer = String::new();
+            reader.read_to_string(&mut buffer)?;
+
+            re.split(&buffer)
+                .map(|s| s.replace('\n', " ")) // ptx behavior: newlines become spaces inside sentences
+                .filter(|s| !s.is_empty()) // remove empty sentences
+                .collect()
+        } else {
+            reader.lines().collect::<std::io::Result<Vec<String>>>()?
+        };
```

Comment:
maybe move that into a function

Reply:
Good point, refactored in [81cc64d](https://github.com/uutils/coreutils/pull/9682/commits/81cc64d1cecd857b8ee2def04e0462b3ac6b9811)